modules/workload-identity/main.tf

```
@@ -89,3 +89,11 @@ resource "google_project_iam_member" "workload_identity_sa_bindings" {
   role    = each.value
   member  = local.gcp_sa_fqn
 }
+
+resource "google_project_iam_member" "workload_identity_sa_bindings_additional_projects" {
+  for_each = toset(distinct(flatten([for project, roles in var.additional_projects : [for role in roles : join("=>", [project, role])]])))
```

Thanks @SudharsaneSivamany 

If feasible, I suggest we try to avoid the join/split paradigm.

If you create a set of objects containing project and role, you should be able to pass those directly and avoid the join/split.  I have not tested/debugged this exact code, but something like this:

```suggestion
  for_each = flatten([for project, roles in var.additional_projects : [for role in roles : {project = project, role = role}]])
```

project = each.value.project
role = each.value.role

---

Sure @apeabody . Changes are made as suggested. Please review.

---

Thanks! Removes the risk of an unexpected value causing an issue with the split.  I've triggered the CI.